### PR TITLE
Add recommendation field and write Plugin_Update recommendation

### LINKED ISSUE #25: Offer a corrective recommendation when a check has a warning or error
When a check has a warning or an error, we should offer a corrective recommendation on what they should do to try to address the issue.

### PR DESCRIPTION
**Pull request by davegaeddert**
@danielbachhuber let me know how you want to tackle this. I added the field and tested it on one check.

Closes #25

### CODE REVIEW

File: inc/checks/class-plugin-update.php [hunk 1]
```diff
@@ -11,19 +11,23 @@ class Plugin_Update extends Plugin {
...
+				$outdated_plugin_names[] = $plugin['name'];
 			}
 		}
+		$update_count = count( $outdated_plugin_names );
+		$outdated_plugin_names = implode( $outdated_plugin_names, ', ' );
 
 		if ( 1 === $update_count ) {
 			$this->set_status( 'warning' );
 			$this->set_message( "1 plugin has an update available." );
+			$this->set_recommendation( "Update the {$outdated_plugin_names} plugin." );
```
Comment (danielbachhuber) [inline inc/checks/class-plugin-update.php]: Sorry I didn't communicate this well-enough originally. The idea is that `doctor` would recommend which WP-CLI command to run next (if possible).

In saying this, we probably need two attributes: `recommendation_message` and `recommendation_command`.

File: inc/class-command.php [hunk 2]
```diff
@@ -20,8 +20,9 @@ class Command {
 	 * ## OPTIONS
```
Comment (danielbachhuber) [inline inc/class-command.php]: This `## OPTIONS` heading should be below the paragraph text. Can you move it?

File: inc/checks/class-plugin-update.php (hunk 1 above)
Comment (davegaeddert) [reply inc/checks/class-plugin-update.php]: I see. Should I try to take a stab at a command and extended message for each so that we can see how that plays out?

Comment (danielbachhuber) [reply inc/checks/class-plugin-update.php]: Go for it!

**Comment by davegaeddert** [comment]
I added a couple examples in the form of tests. But I had another thought on this. What I was looking for originally was a way to get more information on certain failed checks, particularly the ones where the extra detail (like _which_ PHP files were found in uploads) won't fit nicely in a shorter-style summary "message". What if instead of adding 2 more "recommendation" fields, we did this...

Treat the "message" field more like a git commit message. The first line is a summary and should be around 50 (?) characters. After that summary line can be a blank line, and then essentially as much freeform content as you want. It could list out specific plugins/files/themes/etc. in a human readable list, could include reference links, suggestions, etc. By default, only the summary line is shown and maybe a `--verbose` option (`--verbose-message`? `--message-format=full`?) could optionally output _all_ of the content that it may have. So if they have more content they want to present, then they can. And if not, then it just uses the short message that it already has (which in some cases is already enough). That would also make it pretty easy to add to the messages over time instead of worrying about trying to solve them all now so that there aren't blank "recommendation_message" fields. _And_ you could avoid having two `recommendation_.*` fields, which is a little annoying.

I really like the recommended command idea, and think that's definitely worth playing out (separately). I don't think it will work for all of the checks (ex. php-in-upload), but when it does, people could make pretty slick interactive flows using it (or build an interactive resolution flow into doctor-command itself -- "Want to run `wp plugin update akismet` to resolve this? y\n: ").

So, short version: an expanded, multiline "message" field that only shows first line by default + a separate recommended command field when applicable

@danielbachhuber do you have thoughts on this? Let me know if any of that didn't make sense.

**Comment by danielbachhuber** [comment]
> What I was looking for originally was a way to get more information on certain failed checks, particularly the ones where the extra detail (like which PHP files were found in uploads) won't fit nicely in a shorter-style summary "message".

Oh right. I was thinking these could go in a `data` field (or something better named). I think it needs to be in a separate field though, because it could be a huge amount of data and probably wouldn't be something you want to try to parse out of a message.

> I really like the recommended command idea, and think that's definitely worth playing out (separately). I don't think it will work for all of the checks (ex. php-in-upload), but when it does, people could make pretty slick interactive flows using it (or build an interactive resolution flow into doctor-command itself

Correct and correct :)

**Comment by davegaeddert** [comment]
Can you give an example of what would be in the `data` field for the php-in-upload check? Would it be the file paths separated by newlines?

**Comment by danielbachhuber** [comment]
> Can you give an example of what would be in the `data` field for the php-in-upload check? Would it be the file paths separated by newlines?

It would be the file path relative to WordPress root in JSON format. In `--format=table`, the JSON would be rendered within the table. In `--format=json`, it would be true JSON.

**Comment by davegaeddert** [comment]
I'm still planning to work on this (we need it), just been caught up in my own stuff. I opened #136 and will probably start over with a new PR when I have time.
